mlflow/utils/model_utils.py

```diff
@@ -118,6 +118,18 @@ def _validate_and_copy_code_paths(code_paths, path, default_subpath="code"):
 
 def _add_code_to_system_path(code_path):
     sys.path = [code_path] + _get_code_dirs(code_path) + sys.path
+    # delete code modules that are already in sys.modules
+    # so they will get reloaded anew from the correct code path
+    # othterwise python will use the alredy loaded modules
+    modules = []
+    for path in [code_path] + _get_code_dirs(code_path):
+        modules_py = glob.glob(join(path, "*.py"))
+        modules += [
+            basename(f)[:-3] for f in modules_py if isfile(f) and not f.endswith("__init__.py")
+        ]
+    for module in modules:
+        if module in sys.modules:
+            sys.modules.pop(module)
```

Comment:
```suggestion
    for module in modules:
        sys.modules.pop(module, None)
```

Reply:
dict(zip([str(x) for x in list(range(10))], list(range(10)))).pop(100, None) this returns None.

Reply:
@BFAnas It does and removes the key `100` **if exists**.